cmake/inputs/patches/ep_gcssdk/build.patch

```diff
@@ -0,0 +1,389 @@
+From: Joe maley <joe@tiledb.io>
+Date: Wed, 11 Mar 2020 11:26:06 -0400
+Subject: GCS Build Patch for TileDB
+
+---
+ CMakeLists.txt                               |  2 +-
+ cmake/FindCurlWithTargets.cmake              |  1 +
+ super/CMakeLists.txt                         | 13 ++--
+ super/external/c-ares.cmake                  |  2 +
+ super/external/crc32c.cmake                  |  2 +
+ super/external/curl.cmake                    | 64 --------------------
+ super/external/google-cloud-cpp-common.cmake |  2 +
+ super/external/googleapis.cmake              |  4 +-
+ super/external/googletest.cmake              |  2 +
+ super/external/grpc.cmake                    |  7 ++-
+ super/external/protobuf.cmake                |  4 +-
+ super/external/ssl.cmake                     | 26 --------
+ super/external/zlib.cmake                    | 47 --------------
+ 13 files changed, 28 insertions(+), 148 deletions(-)
+ delete mode 100644 super/external/curl.cmake
+ delete mode 100644 super/external/ssl.cmake
+ delete mode 100644 super/external/zlib.cmake
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c9275541b..b15cf2ca2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -53,7 +53,7 @@ endif ()
+ 
+ option(GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK
+        "If enabled, check that the user has defined OPENSSL_ROOT_DIR on macOS"
+-       ON)
++       OFF)
+ if (APPLE)
+     # This is an easy mistake to make, and the error messages are very
+     # confusing. Help our users by giving them some guidance.
+diff --git a/cmake/FindCurlWithTargets.cmake b/cmake/FindCurlWithTargets.cmake
+index 5fe8cbfba..c189c2781 100644
+--- a/cmake/FindCurlWithTargets.cmake
++++ b/cmake/FindCurlWithTargets.cmake
+@@ -31,6 +31,7 @@ else ()
+     # CMake config module. We will need to fix up a few things if the module is
+     # found this way.
+     find_package(CURL REQUIRED)
++
+     # Before CMake 3.12 the module does not define a target, compare:
+     # https://cmake.org/cmake/help/v3.12/module/FindCURL.html vs
+     # https://cmake.org/cmake/help/v3.11/module/FindCURL.html
+diff --git a/super/CMakeLists.txt b/super/CMakeLists.txt
+index 9cc916aed..b5fcd8c54 100644
+--- a/super/CMakeLists.txt
++++ b/super/CMakeLists.txt
+@@ -23,7 +23,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ set(GOOGLE_CLOUD_CPP_ROOT "${PROJECT_SOURCE_DIR}/..")
+ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
+ 
+-include(external/curl)
+ include(external/crc32c)
+ include(external/grpc)
+ include(external/googleapis)
+@@ -35,9 +34,11 @@ set_external_project_build_parallel_level(PARALLEL)
+ set_external_project_vars()
+ 
+ set(GOOGLE_CLOUD_CPP_DEPENDENCIES_LIST
+-    grpc-project curl-project crc32c-project googleapis-project
++    grpc-project crc32c-project googleapis-project
+     googletest-project google-cloud-cpp-common-project)
+ 
++set(CMAKE_PREFIX_PATH "${GOOGLE_CLOUD_CPP_PREFIX_PATH}|${TILEDB_EP_INSTALL_PREFIX}")
```

Comment:
@ihnorton @Shelnutt2 Here is where the `TILEDB_EP_INSTALL_PREFIX` is appended to the `CMAKE_PREFIX_PATH` var. Note line 80 in this file as well.

Reply:
I think you can just set `-DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}` in the EP_Add call. But if this is working we can leave it for now.

Reply:
in *our* `EP_Add` call

Reply:
If we did that, wouldn't we still need a line like this to append `GOOGLE_CLOUD_CPP_PREFIX_PATH` to the `CMAKE_PREFIX_PATH`? Either way works, but `CMAKE_PREFIX_PATH` must have both `TILEDB_EP_INSTALL_PREFIX` and `GOOGLE_CLOUD_CPP_PREFIX_PATH` before _this_ `EP_Add`

Reply:
I believe they construct `GOOGLE_CLOUD_CPP_PREFIX_PATH` from the initial `CMAKE_PREFIX_PATH` and this is passed to the GCS-side EP_Add for storage.

Reply:
Line 79 here indicates the opposite: `CMAKE_PREFIX_PATH` is assigned to `GOOGLE_CLOUD_CPP_PREFIX_PATH`. I've printed `CMAKE_PREFIX_PATH` at this line and it is empty.

Reply:
see https://github.com/googleapis/google-cloud-cpp/blob/50119a35efc97e92fc20fc6369e72feccdeab984/super/ExternalProjectHelper.cmake#L57-L59

Reply:
Good catch, I just changed this to set `CMAKE_PREFIX_PATH`